clientconn.go

```diff
@@ -788,6 +790,87 @@ func (cc *ClientConn) getTransport(ctx context.Context, failfast bool) (transpor
 	return t, done, nil
 }
 
+func parseTimeout(t *string) (*time.Duration, error) {
+	if t == nil {
+		return nil, nil
+	}
+	d, err := time.ParseDuration(*t)
+	return &d, err
+}
+
+type jsonName struct {
+	Service *string `json:"service,omitempty"`
+	Method  *string `json:"method,omitempty"`
+}
+
+func (j jsonName) Stringer() string {
+	res := ""
+	if j.Service != nil {
+		res += "/" + *j.Service + "/"
+	}
+	if j.Method != nil {
+		res += *j.Method
```

Comment:
If `service == nil`, this function returns `"method"`, is this right?

When searching in the map, we search for `/service/method` first, then `/service/`, but never for `method`. Do we need to store the value in the map if `service == nil`?

Reply:
right. change has been made to skip storing such method config.